### PR TITLE
[Optimism] WETH connector fix

### PR DESCRIPTION
**Pull request by thrilok209**
Issue: https://community.optimism.io/docs/developers/known-issues.html#issue-1852-gas-pricing-discrepancy

Fix: 
Create an intermediate contract([InstaWETHWithdraw](https://gist.github.com/thrilok209/49b30080f3ed00f44a99ae1f4716606d)) to support the unwrapping of WETH to DSA wallet.

Todo:
- [x] Deploy InstaWETHWithdraw contract. 
- [x] Update InstaWETHWithdraw contract address on connector
- [x] Deploy WETH connector

### CODE REVIEW

File: contracts/optimism/connectors/weth/main.sol [hunk 1]
```diff
@@ -27,7 +27,7 @@ abstract contract Resolver is Events, DSMath, Basic, Helpers {
         uint _amt = getUint(getId, amt);
 
         _amt = _amt == uint(-1) ? address(this).balance : _amt;
-        wethContract.deposit{value: _amt}();
+        wethFixContract.deposit{value: _amt}();
```
Comment (thrilok209) [inline contracts/optimism/connectors/weth/main.sol]:
```suggestion
        wethContract.deposit{value: _amt}();
```

File: contracts/optimism/connectors/weth/main.sol [hunk 2]
```diff
@@ -50,8 +50,8 @@ abstract contract Resolver is Events, DSMath, Basic, Helpers {
         uint _amt = getUint(getId, amt);
 
         _amt = _amt == uint(-1) ? wethContract.balanceOf(address(this)) : _amt;
-        approve(wethContract, wethAddr, _amt);
-        wethContract.withdraw(_amt);
+        approve(wethFixContract, wethAddr, _amt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/weth/main.sol]:
```suggestion
        approve(wethContract, address(wethContract), _amt);
```

**Comment by pradyuman-verma** [comment]
### Deployed on Optimism 

**Name:** `InstaWETHWithdraw `
**Address:** [0xa5044f8FfA8FbDdd0781cEDe502F1C493BB6978A](https://optimistic.etherscan.io/address/0xa5044f8FfA8FbDdd0781cEDe502F1C493BB6978A#code)

**Name:** `WETH-A`
**Address:** [0x17Ad93BfD041Fc9784c4Fd203330bC4b0320c272](https://optimistic.etherscan.io/address/0x17Ad93BfD041Fc9784c4Fd203330bC4b0320c272#code)
**tx:** [here](https://optimistic.etherscan.io/tx/0xf8d4ebe7f0d012e6ca4a273dba10bd7f1e041f3c9c8b7aa4f9b414f65f2ed2be)

**Review by pradyuman-verma — APPROVED** [review]

File: contracts/optimism/connectors/weth/main.sol [hunk 3]
```diff
@@ -50,8 +50,8 @@ abstract contract Resolver is Events, DSMath, Basic, Helpers {
         uint _amt = getUint(getId, amt);
 
         _amt = _amt == uint(-1) ? wethContract.balanceOf(address(this)) : _amt;
-        approve(wethContract, wethAddr, _amt);
-        wethContract.withdraw(_amt);
+        approve(wethContract, address(wethContract), _amt);
```
Comment (thrilok209) [inline contracts/optimism/connectors/weth/main.sol]:
```suggestion
        approve(wethContract, address(wethFixContract), _amt);
```

**Pull request merged**